src/engine/TransitivePath.cpp

```diff
@@ -146,11 +148,11 @@ VariableToColumnMap TransitivePath::computeVariableToColumnMap() const {
 // _____________________________________________________________________________
 void TransitivePath::setTextLimit(size_t limit) {
   _subtree->setTextLimit(limit);
-  if (_leftSideTree != nullptr) {
-    _leftSideTree->setTextLimit(limit);
+  if (_lhs.treeAndCol.has_value()) {
+    _lhs.treeAndCol.value().first->setTextLimit(limit);
   }
-  if (_rightSideTree != nullptr) {
-    _rightSideTree->setTextLimit(limit);
+  if (_rhs.treeAndCol.has_value()) {
+    _rhs.treeAndCol.value().first->setTextLimit(limit);
```

Comment:
I think `for each child in children -> child->setTextLimit()` also does the job and is more idiomatic
(This should actually be the default implementation of this function, but that should probably be in a different PR.